tools/circleci/images/Dockerfile

```diff
@@ -43,6 +43,13 @@ RUN apt-get update \
   && { locale-gen C.UTF-8 || true; }
 ENV LANG=C.UTF-8
 
+# Set the locale
+RUN locale-gen en_US.UTF-8
+ENV LANG en_US.UTF-8
+ENV LANGUAGE en_US:en
+ENV LC_ALL en_US.UTF-8
+RUN echo "LC_ALL=en_US.UTF-8" | sudo tee -a /etc/default/locale
```

Comment:
Huh, fascinating, thanks for debugging this. A few thoughts to make this cleaner:

* Right above this, we set up the `C.UTF-8` locale. But we don't set it up in all the same ways. I'd like to isolate which change here was important -- if we set up `C.UTF-8` in all these same ways instead, does that solve the problem? Or, if we just replace `C.UTF-8` with `en_US.UTF-8` in the lines above, does that solve the problem?

* Good practice for writing Dockerfiles is to reduce the number of separate RUN instructions -- each one makes a separate layer in the final image, which affects performance. There's a balance to be had here with readability, in not piling *everything* into one instruction, but that's why I've kept related things grouped together with `&&`. So this `locale-gen` and the `/etc/default/locale` thing should go together with the instruction above.

Reply:
Oh, and

* Definitely explain in the commit message what this is fixing, because otherwise it looks kind of magical. :)

Reply:
Also worth reading https://docs.docker.com/engine/userguide/eng-image/dockerfile_best-practices/ and https://docs.docker.com/engine/reference/builder/ , from the upstream docs.

I was pretty sure I saw the thing about fewer RUN instructions explained there before, but I see it only briefly mentioned now, so I might be thinking of somewhere else -- but there's other good info in those pages.

Reply:
For this sort of thing, I generally prefer the explanation for why we do this in a code comment over one in the commit message: those are much more handy.

Reply:
After trying out various combinations I have arrived at a conclusion that all these steps are necessary. All the other combinations results in provision to fail or backend tests to fail.
```
+ENV LANG en_US.UTF-8
+ENV LANGUAGE en_US:en
+ENV LC_ALL en_US.UTF-8
+RUN echo "LC_ALL=en_US.UTF-8" | sudo tee -a /etc/default/locale
```

Reply:
Hmm. Can you say exactly what versions you tried?

One reason I'm a bit skeptical is, LANG has no effect at all when LC_ALL is set:
https://www.gnu.org/software/gettext/manual/html_node/Locale-Environment-Variables.html

And the only effect of that `LANGUAGE` setting should be to fall back to `en` if `en_US` is not available, which doesn't seem likely to be our issue.